django_auth_adfs/backend.py

```diff
@@ -172,22 +173,25 @@ def update_user_attributes(self, user, claims):
 
         required_fields = [field.name for field in user._meta.fields if field.blank is False]
 
-        for field, claim in settings.CLAIM_MAPPING.items():
-            if hasattr(user, field):
-                if claim in claims:
-                    setattr(user, field, claims[claim])
-                    logger.debug("Attribute '%s' for user '%s' was set to '%s'.", field, user, claims[claim])
-                else:
-                    if field in required_fields:
-                        msg = "Claim not found in access token: '{}'. Check ADFS claims mapping."
-                        raise ImproperlyConfigured(msg.format(claim))
+        for field, claim in mapped_fields.items():
+            if type(claim) != dict:
+                if hasattr(user, field):
+                    if claim in claims:
+                        setattr(user, field, claims[claim])
+                        logger.debug("Attribute '%s' for user '%s' was set to '%s'.", field, user, claims[claim])
                     else:
-                        logger.warning("Claim '%s' for user field '%s' was not found in "
-                                       "the access token for user '%s'. "
-                                       "Field is not required and will be left empty", claim, field, user)
+                        if field in required_fields:
+                            msg = "Claim not found in access token: '{}'. Check ADFS claims mapping."
+                            raise ImproperlyConfigured(msg.format(claim))
+                        else:
+                            logger.warning("Claim '%s' for user field '%s' was not found in "
+                                        "the access token for user '%s'. "
+                                        "Field is not required and will be left empty", claim, field, user)
+                else:
+                    msg = "User model has no field named '{}'. Check ADFS claims mapping."
+                    raise ImproperlyConfigured(msg.format(field))
```

Comment:
This check needs to happen in the outer loop to prevent the `else` branch below from causing an error in `getattr(user, field)` when it doesn't exist.

Reply:
Done: https://github.com/snok/django-auth-adfs/pull/242/commits/91f7ee5273cacda41e7a790e378bd8698fc3f39a